gplugins/design_recipe/DesignRecipe.py

```diff
@@ -1,81 +1,151 @@
 from __future__ import annotations
 
-from typing import List, Optional
+import json
+import os
+import pickle
+from collections.abc import Callable
+from pathlib import Path, PosixPath, WindowsPath
+
+import pydantic
 from gdsfactory import Component
-from gdsfactory.pdk import LayerStack, get_layer_stack
+from gdsfactory.config import logger
 from gdsfactory.path import hashlib
+from gdsfactory.pdk import LayerStack, get_layer_stack
+from gdsfactory.typings import ComponentFactory
+from pydantic import BaseModel
 
 import gplugins.design_recipe as dr
 
 
+class Setup:
+    """
+    A dynamic class to store any setup information
+
+    This class allows designers to arbitrarily add setup information.
+    """
+
+    def __init__(self, **kwargs):
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    def __eq__(self, other):
+        if not isinstance(other, Setup):
+            # Don't attempt to compare against unrelated types
+            return NotImplemented
```

Comment:
should this be a `raise` (perhaps with a more descriptive error message)?

Reply:
No need, we only want to compare Setup instances with one another, no other types.